pkg/lock/sortable_mutex.go

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package lock
+
+import (
+	"sort"
+	"sync"
+	"sync/atomic"
+)
+
+// sortableMutexSeq is a global sequence counter for the creation of new
+// SortableMutex's with unique sequence numbers.
+var sortableMutexSeq atomic.Uint64
+
+// SortableMutex provides a Mutex that can be globally sorted with other
+// sortable mutexes. This allows deadlock-safe locking of a set of mutexes
+// as it guarantees consistent lock ordering.
+type SortableMutex struct {
```

Comment:
Since that `sync.Mutex`es are usable by default, while this isn't, I think the struct should not be exported.

Reply:
Ah good catch. I hid the struct behind an inteface, so it cannot now be initialized wrong.